### PR TITLE
[SQL]: az sql mi op list, az sql mi op get, az sql mi op cancel

### PR DESCRIPTION
**Pull request by toki95**
**History Notes:**  
(Fill in the following template if multiple notes are needed, otherwise PR title will be used for history note.)

[Component Name 1] (BREAKING CHANGE:) (az command:) make some customer-facing change.  
[Component Name 2] (BREAKING CHANGE:) (az command:) make some customer-facing change.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S167

File: src/azure-cli/azure/cli/command_modules/sql/_util.py [hunk 1]
```diff
@@ -36,6 +36,10 @@ def get_sql_database_operations_operations(cli_ctx, _):
     return get_sql_management_client(cli_ctx).database_operations
 
 
+def get_sql_managed_instance_operations_operations(cli_ctx, _):
```
Comment (jaredmoo) [inline src/azure-cli/azure/cli/command_modules/sql/_util.py]: Please arrange alphabetically

File: src/azure-cli/azure/cli/command_modules/sql/commands.py [hunk 2]
```diff
@@ -27,6 +27,7 @@
     get_sql_database_blob_auditing_policies_operations,
     get_sql_database_sensitivity_labels_operations,
     get_sql_database_operations_operations,
+    get_sql_managed_instance_operations_operations,
```
Comment (jaredmoo) [inline src/azure-cli/azure/cli/command_modules/sql/commands.py]: Please arrange alphabetically

File: src/azure-cli/azure/cli/command_modules/sql/commands.py [hunk 3]
```diff
@@ -184,6 +185,16 @@ def load_command_table(self, _):
         g.command('list', 'list_by_database')
         g.command('cancel', 'cancel')
 
+    managed_instance_operations_operations = CliCommandType(
+        operations_tmpl='azure.mgmt.sql.operations#ManagedInstanceOperations.{}',
+        client_factory=get_sql_managed_instance_operations_operations)
+
+    with self.command_group('sql mi op', managed_instance_operations_operations) as g:
```
Comment (jaredmoo) [inline src/azure-cli/azure/cli/command_modules/sql/commands.py]: Please arrange alphabetically

File: src/azure-cli/azure/cli/command_modules/sql/_help.py [hunk 4]
```diff
@@ -163,6 +163,18 @@
     text: az sql db op cancel -g mygroup -s myserver -d mydb -n d2896mydb-2ba8-4c84-bac1-387c430cce40
 """
 
+helps['sql mi op'] = """
```
Comment (jaredmoo) [inline src/azure-cli/azure/cli/command_modules/sql/_help.py]: Please arrange alphabetically

File: src/azure-cli/azure/cli/command_modules/sql/_params.py [hunk 5]
```diff
@@ -702,6 +702,20 @@ def load_arguments(self, _):
...
+    with self.argument_context('sql mi op') as c:
+        c.argument('managed_instance_name',
+                   options_list=['--managed-instance', '--mi'],
+                   required=True,
+                   help='Name of the Azure SQL Managed Instance.')
+
+        c.argument('operation_id',
+                   options_list=['--name', '-n'],
+                   required=True,
+                   help='The unique name of the operation to cancel.')
```
Comment (jaredmoo) [inline src/azure-cli/azure/cli/command_modules/sql/_params.py]: Does this help text make sense for `op show`?

**Comment by toki95** [comment]
The checks are failing probably due to the fact that clients are not merged yet:
https://github.com/Azure/azure-sdk-for-python/pull/10316/files

**Comment by jaredmoo** [comment]
Why did you put `BREAKING CHANGE` in the title?

**Comment by toki95** [comment]
> Why did you put `BREAKING CHANGE` in the title?

I didn't understood the guidance fully the first time I read it, I thought that this can not be empty and that it can be either braking change or hotfix. I have removed it.

File: src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py [hunk 6]
```diff
@@ -711,6 +711,74 @@ def test_sql_db_operation_mgmt(self, resource_group, resource_group_location, se
...
+    def test_sql_mi_operation_mgmt(self):
+        managed_instance_name = self.create_random_name(managed_instance_name_prefix, managed_instance_name_max_length)
+        admin_login = 'admin123'
+        admin_password = 'SecretPassword123'
+
+        is_playback = os.path.exists(self.recording_file)
+        if is_playback:
+            subnet = '/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/AndyPG/providers/Microsoft.Network/virtualNetworks/prepare-cl-nimilj/subnets/default'
+        else:
+            subnet = '/subscriptions/a8c9a924-06c0-4bde-9788-e7b1370969e1/resourceGroups/AndyPG/providers/Microsoft.Network/virtualNetworks/prepare-cl-nimilj/subnets/default'
```
Comment (jaredmoo) [inline src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py]: Please do not hardcode resources, it causes massive problems for other engineers when they need to re-record your test. Follow example of other MI CLI tests that create subnet at runtime.

File: src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py [hunk 7]
```diff
@@ -711,6 +711,74 @@ def test_sql_db_operation_mgmt(self, resource_group, resource_group_location, se
...
+        license_type = 'LicenseIncluded'
+        loc = 'eastus2euap'
+        v_cores = 8
+        storage_size_in_gb = '128'
+        edition = 'GeneralPurpose'
+        family = 'Gen5'
+        resource_group = "DejanDuVnetRG"
+        user = admin_login
+
+        print('Creating MI...\n')
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py]: I recommend using logger.

File: src/azure-cli/azure/cli/command_modules/sql/commands.py [hunk 8]
```diff
@@ -490,6 +491,16 @@ def load_command_table(self, _):
...
     ###############################################
 
+    managed_instance_operations_operations = CliCommandType(
+        operations_tmpl='azure.mgmt.sql.operations#ManagedInstanceOperations.{}',
+        client_factory=get_sql_managed_instance_operations_operations)
+
+    with self.command_group('sql mi op', managed_instance_operations_operations) as g:
+
+        g.command('list', 'list_by_managed_instance')
+        g.command('show', 'get')
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/sql/commands.py]: Consider using `g.show_command()`

File: src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py (hunk 6 above)
Comment (toki95) [reply src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py]: Sure, will change.

**Pull request merged**